lib/common.js

```diff
@@ -545,6 +574,7 @@ exports.readMysqlValue = function(parser, column, columnSchema) {
     case MysqlTypes.DECIMAL: // Deprecated in MySQL > 5.0.3
     case MysqlTypes.NEWDATE: // Not used
     default:
+      // TODO: Remove throw statement to make module safe
```

Comment:
A parsing error doesn't need to create an error or ``throw``... when parsing a row we can accumulate any parsing errors and then emit them all at once. This may require refactoring and a tighter coupling of row processing to column processing, however, every column is in a row, so making the piece of work a row, which is a set of columns, which is a set of values that are different data types, the structure is still sound.